src/backend/distributed/commands/type.c

```diff
@@ -176,9 +159,36 @@ PostprocessCompositeTypeStmt(Node *node, const char *queryString)
 	 * locally it can't be missing
 	 */
 	ObjectAddress typeAddress = GetObjectAddressFromParseTree(node, false);
+
+	/* If the type has any unsupported dependency, create it locally */
+	DeferredErrorMessage *errMsg = DeferErrorIfHasUnsupportedDependency(&typeAddress);
+	if (errMsg != NULL)
+	{
+		RaiseDeferredError(errMsg, WARNING);
+		return NIL;
+	}
+
 	EnsureDependenciesExistOnAllNodes(&typeAddress);
 
-	return NIL;
+	/* fully qualify before lookup and later deparsing */
+	QualifyTreeNode(node);
```

Comment:
QualifyTreeNode must always happen in preprocess

The only reason we do this is to ensure that the schema that's used is going to be the same between our code and the process utility hook, regardless of (concurrent) search_path changes.

Reply:
@agedemenli can you please check whether we need to update aggregate code as well? As I remember we've introduced only postprocess function for aggregates.

Reply:
Yes, we have QualifyNode in Postprocess phase, for aggregates.
I'm going to remove that, then add a Preprocess function for CREATE AGGREGATE statements, which only does a QualifyNode and returns NIL.

@velioglu @marcocitus